### PR TITLE
Remove the remaining TODOs

### PR DESCRIPTION
**Pull request by tcharding**
Add issues and remove the TODOs from the code.

Resolves: #2368

### CODE REVIEW

**Comment by tcharding** [comment]
If this merges we could consider enforcing no TODOs in CI and start using issues exclusively?

File: bitcoin/src/blockdata/weight.rs [hunk 1]
```diff
@@ -63,11 +63,9 @@ impl Weight {
         match vb.checked_mul(Self::WITNESS_SCALE_FACTOR) {
             Some(weight) => Weight(weight),
             None => {
-                // TODO replace with panic!() when MSRV = 1.57+
+                // When MSRV is greater than 1.57 we can use `panic!()`.
```
[the remark this answers is not in the document]
Comment (apoelstra) [reply bitcoin/src/blockdata/weight.rs:66]: We should change this before merging. We'll definitely forget this conversation :).

[the remark this answers is not in the document]
Comment (yancyribbens) [reply bitcoin/src/blockdata/weight.rs:66]: `1.57+` means 1.57 or greater I think.

**Comment by tcharding** [comment]
Fixed the version comment to use `// When MSRV is 1.57+ we can use `panic!()`.` Sloppy work, my bad.

**Review by apoelstra — APPROVED** [review]
ACK c69caafefca08453dbc13a64c5933f7efe209272

File: bitcoin/src/blockdata/weight.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kixunil) [reply bitcoin/src/blockdata/weight.rs]: @apoelstra we have this in the tracking issue and we could actually bump MSRV to that version right now - it's within our policy. So it's not a big deal that the comment is not super-accurate.

@yancyribbens yes it does and it was correct, it was changed to words which don't say that and are thus not correct but not a big deal.

File: units/src/amount.rs [hunk 2]
```diff
@@ -716,15 +716,12 @@ impl Amount {
     /// The function panics if the argument multiplied by the number of sats
     /// per bitcoin overflows a u64 type.
     pub const fn from_int_btc(btc: u64) -> Amount {
-        // TODO replace whith unwrap() when available in const context.
         match btc.checked_mul(100_000_000) {
             Some(amount) => Amount::from_sat(amount),
             None => {
-                // TODO replace with panic!() when MSRV = 1.57+
+                // When MSRV is greater than 1.57 we can use `panic!()`.
```
[the remark this answers is not in the document]
Comment (Kixunil) [reply units/src/amount.rs]: We can't because our MSRV is 1.56.

**Review by Kixunil — APPROVED** [review]
ACK c69caafefca08453dbc13a64c5933f7efe209272

**Pull request merged**